cider/common/base/wide_integer.h

```diff
@@ -0,0 +1,327 @@
+                                                                const Arithmetic2& lhs);
+
+template <size_t Bits, typename Signed, size_t Bits2, typename Signed2>
+std::common_type_t<integer<Bits, Signed>, integer<Bits2, Signed2>> constexpr operator/(
+    const integer<Bits, Signed>& lhs,
+    const integer<Bits2, Signed2>& rhs);
+template <typename Arithmetic,
+          typename Arithmetic2,
+          class = _only_arithmetic<Arithmetic, Arithmetic2>>
+std::common_type_t<Arithmetic, Arithmetic2> constexpr operator/(const Arithmetic& rhs,
+                                                                const Arithmetic2& lhs);
+
+template <size_t Bits, typename Signed, size_t Bits2, typename Signed2>
+std::common_type_t<integer<Bits, Signed>, integer<Bits2, Signed2>> constexpr operator+(
+    const integer<Bits, Signed>& lhs,
+    const integer<Bits2, Signed2>& rhs);
+template <typename Arithmetic,
+          typename Arithmetic2,
+          class = _only_arithmetic<Arithmetic, Arithmetic2>>
+std::common_type_t<Arithmetic, Arithmetic2> constexpr operator+(const Arithmetic& rhs,
+                                                                const Arithmetic2& lhs);
+
+template <size_t Bits, typename Signed, size_t Bits2, typename Signed2>
+std::common_type_t<integer<Bits, Signed>, integer<Bits2, Signed2>> constexpr operator-(
+    const integer<Bits, Signed>& lhs,
+    const integer<Bits2, Signed2>& rhs);
+template <typename Arithmetic,
+          typename Arithmetic2,
+          class = _only_arithmetic<Arithmetic, Arithmetic2>>
+std::common_type_t<Arithmetic, Arithmetic2> constexpr operator-(const Arithmetic& rhs,
+                                                                const Arithmetic2& lhs);
+
+template <size_t Bits, typename Signed, size_t Bits2, typename Signed2>
+std::common_type_t<integer<Bits, Signed>, integer<Bits2, Signed2>> constexpr operator%(
+    const integer<Bits, Signed>& lhs,
+    const integer<Bits2, Signed2>& rhs);
+template <typename Integral,
+          typename Integral2,
+          class = _only_integer<Integral, Integral2>>
+std::common_type_t<Integral, Integral2> constexpr operator%(const Integral& rhs,
+                                                            const Integral2& lhs);
+
+template <size_t Bits, typename Signed, size_t Bits2, typename Signed2>
+std::common_type_t<integer<Bits, Signed>, integer<Bits2, Signed2>> constexpr operator&(
+    const integer<Bits, Signed>& lhs,
+    const integer<Bits2, Signed2>& rhs);
+template <typename Integral,
+          typename Integral2,
+          class = _only_integer<Integral, Integral2>>
+std::common_type_t<Integral, Integral2> constexpr operator&(const Integral& rhs,
+                                                            const Integral2& lhs);
+
+template <size_t Bits, typename Signed, size_t Bits2, typename Signed2>
+std::common_type_t<integer<Bits, Signed>, integer<Bits2, Signed2>> constexpr operator|(

```

Comment:
I see there are a lot of operator rewrite against the data types, will there be any chance for us to use it?